pkg/interop/interop.go

```diff
@@ -1,58 +1,63 @@
 package interop
 
 import (
-	"context"
-	"fmt"
-	"os"
-	"time"
-
-	"github.com/newrelic/go-agent/v3/integrations/logcontext-v2/nrlogrus"
-	"github.com/newrelic/go-agent/v3/newrelic"
-	nrClient "github.com/newrelic/newrelic-client-go/newrelic"
-	"github.com/newrelic/newrelic-client-go/pkg/config"
-	"github.com/newrelic/newrelic-client-go/pkg/logging"
-	"github.com/newrelic/newrelic-client-go/pkg/region"
-	log "github.com/sirupsen/logrus"
-	"github.com/spf13/viper"
+  "context"
+  "fmt"
+  "os"
+  "time"
+
+  "github.com/newrelic/go-agent/v3/integrations/logcontext-v2/nrlogrus"
+  "github.com/newrelic/go-agent/v3/newrelic"
+  nrClient "github.com/newrelic/newrelic-client-go/newrelic"
+  "github.com/newrelic/newrelic-client-go/pkg/config"
+  "github.com/newrelic/newrelic-client-go/pkg/logging"
+  "github.com/newrelic/newrelic-client-go/pkg/region"
+  log "github.com/sirupsen/logrus"
+  "github.com/spf13/viper"
 )
 
 type Interop struct {
-  App               *newrelic.Application
-  Logger            *log.Logger
-  NrClient          *nrClient.NewRelic
-  eventsEnabled     bool
+  App           *newrelic.Application
+  Logger        *log.Logger
+  NrClient      *nrClient.NewRelic
+  eventsEnabled bool
 }
 
 func ConfigLicenseKey(licenseKey string) nrClient.ConfigOption {
-	return func(cfg *config.Config) error {
-		cfg.LicenseKey = licenseKey
-		return nil
-	}
+  return func(cfg *config.Config) error {
+    cfg.LicenseKey = licenseKey
+    return nil
+  }
 }
 
 func NewInteroperability() (*Interop, error) {
-  app, err := newrelic.NewApplication(
-    newrelic.ConfigAppName("New Relic Entity Tag Sync"),
-    newrelic.ConfigLicense(os.Getenv("NEW_RELIC_LICENSE_KEY")),
-  )
-  if err != nil {
-		return nil, err
-  }
-
-  logger := log.New()
-
-  logger.SetLevel(log.WarnLevel)
-  logger.SetFormatter(nrlogrus.NewFormatter(app, &log.TextFormatter{}))
-
+  // Load configuration with viper
   viper.SetConfigName("config")
   viper.AddConfigPath("configs")
   viper.AddConfigPath(".")
-
-  err = viper.ReadInConfig()
+  err := viper.ReadInConfig()
   if err != nil {
     return nil, err
   }
 
+  // Look for our license key
+  licenseKey := viper.GetString("licenseKey")
+  if licenseKey == "" {
+    licenseKey = os.Getenv("NEW_RELIC_LICENSE_KEY")
+  }
```

Comment:
Correct me if I'm wrong: this looks like the precedence order isn't handled correctly here. This will only grab license key from the env if it's _not_ in the config file. However, if `NEW_RELIC_LICENSE_KEY` is set as an env var, we want it to _override_ the licenseKey from the config file.

Reply:
We're not using `viper` that way here, we're doing a one off just for this so we don't have to touch a bunch of files

Reply:
We do use it that way [inside the SNOW provider](https://github.com/newrelic/nr-entity-tag-sync/blob/main/internal/provider/servicenow/servicenow.go#L51).  I'm not sure why I didn't do it globally.  And that is not worth changing and wouldn't make sense as part of this PR.